ocaml-lsp-server/src/dune.ml

```diff
@@ -366,28 +366,40 @@ end = struct
       | `Unix s -> Unix.ADDR_UNIX s
       | `Ip (`Host h, `Port p) -> Unix.ADDR_INET (Unix.inet_addr_of_string h, p)
     in
+    let sock =
+      let domain = Unix.domain_of_sockaddr sockaddr in
+      let socket = Unix.socket ~cloexec:true domain Unix.SOCK_STREAM 0 in
+      Lev_fiber.Fd.create socket (`Non_blocking false)
+    in
     let* session =
-      let sock =
-        let domain = Unix.domain_of_sockaddr sockaddr in
-        let socket = Unix.socket ~cloexec:true domain Unix.SOCK_STREAM 0 in
-        Lev_fiber.Fd.create socket (`Non_blocking false)
-      in
       Fiber.map_reduce_errors
         (module Monoid.List (Exn_with_backtrace))
         ~on_error:(fun exn -> Fiber.return [ exn ])
         (fun () -> Csexp_rpc.connect sock sockaddr)
     in
     match session with
-    | Error exns ->
-      let message =
-        let exn = List.hd exns in
-        Format.asprintf "unable to connect to dune %s@.%a"
-          (Registry.Dune.root source)
-          Exn_with_backtrace.pp_uncaught exn
+    | Error exns -> (
+      Lev_fiber.Fd.close sock;
+      let exns =
+        List.filter exns ~f:(fun exn ->
+            match exn with
+            | { Exn_with_backtrace.exn = Unix.Unix_error (ECONNREFUSED, _, _)
```

Comment:
For my own curiosity: why do we want to filter out this error?

Reply:
The error occurs when we fail to connect to a dune instance (because it's not running) usually. It occurs too frequently, so if we log it, we end up flooding the logs with this.